libdnf/dnf-repo.cpp

```diff
@@ -1122,6 +1122,9 @@ dnf_repo_setup(DnfRepo *repo, GError **error)
 {
     DnfRepoPrivate *priv = GET_PRIVATE(repo);
     DnfRepoEnabled enabled = DNF_REPO_ENABLED_NONE;
+    g_autoptr(GDir) yum_vars_dir = NULL;
+    const gchar *yum_vars_path = "/etc/yum/vars";
```

Comment:
This needs to be configurable, and we should default to `/etc/dnf/vars`.

Reply:
That doesn't mean we should let crap go in now. Also, the main `dnf` program already reads `/etc/dnf/vars`.

Reply:
I'll mark this PR as WIP while I work to add a configuration option.

Reply:
It's `/etc/yum.repos.d` though and *that's* not going to change. 

In the end I guess parse both?

It's also probably worth noting that while both Conan-Kudo and I work in the surrounding ecosystem, neither of us are "officially" maintainers of libdnf currently (though we have [a fork](https://github.com/projectatomic/libdnf) I help co-maintain for rpm-ostree) - so bear that in mind with our reviews.

Reply:
DNF actually already has different behavior for repos anyway. It reads a number of locations, in a given priority order, because YUM upstream and Fedora downstream actually have diverged for many years on the location for repo files. The reasoning for it is a bit unclear, but as a consequence, DNF handles repos.d differently from YUM.